includes/class-wc-cart-totals.php

```diff
@@ -637,7 +637,7 @@ protected function calculate_item_totals() {
 			}
 
 			if ( $this->calculate_tax && $item->product->is_taxable() ) {
-				$total_taxes     = WC_Tax::calc_tax( $item->total, $item->tax_rates, $item->price_includes_tax );
+				$total_taxes     = apply_filters( 'woocommerce_calculate_item_totals_taxes', WC_Tax::calc_tax( $item->total, $item->tax_rates, $item->price_includes_tax ), $item );
```

Comment:
Perhaps worth passing $this through to the filter as well?

Reply:
That, or another idea that I had was to move this filter on line 649 right before setting cart contents. Not sure which is better, if you have a preference? The second approach would be more generic

Reply:
I think it is better where it is now, if you add a filter further down you make $item->total_tax not match up to the total item taxes, in which case you will need to have another filter for that as well.